backend/routes/nft.js

```diff
@@ -25,4 +42,44 @@ router.post('/borrow', checkToken, async (req, res) => {
   res.json(result);
 });
 
+router.get('/', checkToken, async (_, res) => {
+  const nftList = await biddingService.showAll();
+
+  const list = (
+    await Promise.all(
+      nftList.map((item) => {
+        return axios
+          .get(
+            `${tokenHistoryApiUrl}/v2/contract/nft/${item.contractAddr}/token/${item.tokenId}`,
+            option,
+          )
+          .then((_res) => ({
+            endDate: item.endDate,
+            principal: item.principal,
+            interests: item.interests,
+            tokenUri: _res.data.tokenUri,
+          }));
+      }),
+    )
+  ).map((item) => {
+    return axios.get(item.tokenUri).then((r) => ({
+      data: r.data,
+      principal: item.principal,
+      interests: item.interests,
+      endDate: item.endDate,
+    }));
+  });
+
+  const results = (await Promise.all(list)).map((item) => ({
+    src: item.data.image,
+    title: item.data.name,
+    desc: item.data.description,
+    principal: item.principal,
+    interests: item.interests,
+    endDate: item.endDate,
+  }));
+
+  res.json(results);
```

Comment:
현재 49line에서 Promise.all로 받아온 결과 값을 64line map에서 다시 Promise 배열로 응답하고 73line에서 Promise.all을 재실행하고 있는데요.

이렇게 되면 첫번째 Promise.all이 모두 끝난 뒤에 두번째 Promise.all을 실행하기 때문에 첫번째 Promise.all 내에서 먼저 끝난 비동기 작업은 다른 작업을 대기하는 상황이지 않을까 싶습니다?! 🤔 

아래처럼 해보는건 어떨까요?!

- 처음 Promise.all 내 각 Promise들은 fulfilled 되는대로 바로 해당 nft 상세 정보를 가져오는 api를 호출 및 해당 호출을 Promise로 return
- nft 상세 정보까지 가져오는 2번째 Promise 비동기 작업까지 모두 완료되면 Promise.all이 완료되어 result를 세팅

```suggestion
const result = await Promise.all(
    nftList.map((item) => {
      return axios
        .get(
          `${tokenHistoryApiUrl}/v2/contract/nft/${item.contractAddr}/token/${item.tokenId}`,
          option,
        )
        .then(({ data: { tokenUri } }) => {
          return axios.get(tokenUri).then(({ data }) => {
            return {
              src: data.image,
              title: data.name,
              desc: data.description,
              principal: item.principal,
              interests: item.interests,
              endDate: item.endDate,
            };
          });
        });
    }),
  );

  res.json(result);
```

Reply:
@giseoplee 네 좋은 의견 감사합니다. 리뷰에 남겨주신 코드를 토대로 한번 작업해보겠습니다.!! 👍